src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -68,6 +68,10 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
 
         public IEnumerable<TypeProvider> Models => SchemaMap.Values;
 
+        public ResourceGroupExtensions ResourceGroupExtensions => new ResourceGroupExtensions(_context);
+
+        public Dictionary<OperationGroup, Resource> ResourceGroupExtensionResource => EnsureArmResource();
```

Comment:
Should it be IEnumerable list? Also you used EnsureArmResource, I assume that there is a TODO to filter out resources which only have RG as parent later?

Reply:
The parent resource group filter is [here](https://github.com/Azure/autorest.csharp/blob/219eb42ef7524790c377c0dc4be48e2967747b4b/src/AutoRest.CSharp/Mgmt/Generation/ResourceGroupExtensionsWriter.cs#L28)